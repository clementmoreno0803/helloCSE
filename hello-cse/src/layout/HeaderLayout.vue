<template>
  <header>
    <nav>
      <router-link to="/"><h1>STREAM</h1></router-link>
      <search-input></search-input>
    </nav>
    <router-view />
  </header>
</template>

<script setup lang="ts">
import searchInput from "@/components/search-input.vue";
</script>

<style lang="scss" scoped>
h1 {
  @include title(2vh, 1.2rem);
}
nav {
  position: sticky;
  top: 0;
  z-index: 2 * $z-top;
  padding: 2rem;
  display: flex;
  background: $background-black;
  align-items: center;
  a {
    font-weight: $bold;
    color: white;
    opacity: 0.6;
    text-decoration: none;
  }
}
</style>
