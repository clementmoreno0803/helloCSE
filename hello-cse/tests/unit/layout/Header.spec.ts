import { mount } from "@vue/test-utils";
import { createRouter, createWebHistory } from "vue-router";
import { describe, it, expect } from "vitest";
import HeaderLayout from "@/layout/HeaderLayout.vue";
import { createPinia, setActivePinia } from "pinia";

const routes = [{ path: "/", component: { template: "<div>Home</div>" } }];

const router = createRouter({
  history: createWebHistory(),
  routes
});

const generateWrapper = () => {
  const pinia = createPinia();
  return mount(HeaderLayout, {
    global: {
      plugins: [router, setActivePinia(pinia)]
    }
  });
};
describe("HeaderComponent", () => {
  it("should render correctly", () => {
    const wrapper = generateWrapper();

    expect(wrapper.element).toMatchSnapshot();
  });
  it("renders the header and nav elements", () => {
    const wrapper = generateWrapper();

    const title = wrapper.find("h1");
    expect(title.exists()).toBe(true);
    expect(title.text()).toBe("HELLO CSE");

    const searchInput = wrapper.findComponent({ name: "search-input" });
    expect(searchInput.exists()).toBe(true);
  });

  it("navigates to the home page when the title is clicked", async () => {
    const wrapper = generateWrapper();

    await wrapper.find("h1").trigger("click");

    expect(wrapper.vm.$route.path).toBe("/");
  });
});
