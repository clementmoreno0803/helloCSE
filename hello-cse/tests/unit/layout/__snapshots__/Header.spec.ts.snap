// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HeaderComponent > should render correctly 1`] = `
<header
  data-v-1580a921=""
>
  <nav
    data-v-1580a921=""
  >
    <a
      class=""
      data-v-1580a921=""
      href="/"
    >
      <h1
        data-v-1580a921=""
      >
        STREAM
      </h1>
    </a>
    <div
      class="search-bar"
      data-v-02b2df00=""
      data-v-1580a921=""
    >
      <input
        class="search-input"
        data-testid="input-search"
        data-v-02b2df00=""
        placeholder="Veuillez renseigner un film"
        style="width: 2rem;"
        type="text"
      />
    </div>
  </nav>
  <!---->
</header>
`;
